move: variables_exercise (#62)
CodeYourFuture/fundamentals-blockly: exercises/button_variables_state.md

## Variables

> Suppose we have a value that represents the number of times a user clicks on a given button. Every time a user clicks on this button then this value increases by 1. We need to refer to this value so we can check it and update it. 

Author: I completely re-worked this bit - let me know what you think.

> Suppose we have a value that represents the number of times a user clicks on a given button. Every time a user clicks on this button then this value increases by 1. We need to refer to this value so we can check it and update it. 

Reviewer: ```suggestion
Suppose we have a value that represents the number of times a user clicks on a given button. Every time a user clicks on this button then this value increases by 1. We will need to keep track of this value so that we can update it. 
```

We can give this value a label like `click_count` and refer to `click_count` when we need to get the number of clicks. In programming, we call this label a variable. 

> ℹ️ A variable is a label for some value.

Reviewer: ```suggestion
ℹ️ A variable is a label for some value.
ℹ️ A variable that changes value during the running of a program is called "state".
```

Author: Will this definition of state confuse students later on when there is more than one variable? Should we say "state" is any data that gets updated during the running of the program ( this now encompasses variable(s) )

Reviewer: That works for me
